File: code/cpp/ch16/practice/challenge_01_aggressive_cows.cpp
/*
 * Challenge 1: Aggressive Cows
 * Chapter 16: Binary Search Beyond Arrays — Searching on Answers
 *
 * PROBLEM: Place c cows in stalls to maximize minimum distance between any two.
 */

#include <algorithm>
#include <iostream>
#include <vector>
using namespace std;

int solve(vector<int> stalls, int cows) {
    // TODO: Replace this with your solution
    return 0;
}

int main() {
    int n;
    cin >> n;
    vector<int> stalls(n);
    for (int i = 0; i < n; i++) cin >> stalls[i];
    int cows;
    cin >> cows;
    cout << solve(stalls, cows) << endl;
    return 0;
}

File: code/java/ch16/practice/Challenge01AggressiveCows.java
package ch16.practice;

import java.util.*;

/**
 * Challenge 1: Aggressive Cows
 * Chapter 16: Binary Search Beyond Arrays — Searching on Answers
 *
 * PROBLEM: Place c cows in stalls to maximize the minimum distance
 *          between any two cows.
 *
 * INSTRUCTIONS: Replace the body of solve() with your solution.
 */
public class Challenge01AggressiveCows {
    public static int solve(int[] stalls, int cows) {
        // TODO: Replace this with your solution
        return 0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String line = sc.nextLine().trim();
        int[] stalls = Arrays.stream(line.split(" ")).mapToInt(Integer::parseInt).toArray();
        int cows = sc.nextInt();
        System.out.println(solve(stalls, cows));
        sc.close();
    }
}
